Automattic/jetpack
# Voice to Content: Support canceling requests

Fixes #35549. Part of #35547.

## Proposed changes:

* Add support to cancel transcription requests and stop post-processing requests on the respective hooks
* Remove the dependency on the CancelablePromise type
* When the cancel handler is called on the block (because the user clicked the "Cancel" button), cancel a possible transcription and post-processing request that may be going on

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Go to the block editor and open the browser development console; open the network tab to inspect the requests
* Insert a Voice-to-Content block
* Begin the recording of some audio, just for testing purposes
* Click Done and, as soon as you see the transcription request pop on the network tab, click the Cancel button
   * The UI on the block should return to the initial one, and the request should be canceled on the network tab
   * The same should happen when you pick a file for transcription: as soon as the transcription starts, when you click Cancel, the UI should return to the initial state and the network request should be canceled
* Now let's test canceling after the transcription started to be included on the editor
* Record an audio (or pick a file for upload), wait for the transcription to finish, and when you see that the transcription started to appear on the editor (or on the console logs), click Cancel
   * The transcription should stop, leaving you with the initial UI on the block
* For this last test, it's helpful if the audio is long enough to take some time to be inserted on the editor, giving you time to react to it
